src/languageserver/telemetry.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Connection } from 'vscode-languageserver/lib/common/server';
+
+/**
+ * Doe to LSP limitation this object must be JSON serializable
+ */
+export interface TelemetryEvent {
+  type?: string;
+  name: string;
+  properties?: unknown;
+  measures?: unknown;
+  traits?: unknown;
+  context?: unknown;
+}
+
+export class Telemetry {
+  constructor(private readonly connection: Connection) {}
+
+  send(event: TelemetryEvent): void {
+    this.connection.telemetry.logEvent(event);
+  }
+
+  sendError(name: string, properties: unknown): void {
+    this.send({ name, type: 'track', properties });
```

Comment:
Should this be:
```
this.send({ name, type: 'error', properties });
```
?
Currently it looks like this line and sendTrack are the same

Reply:
no, this is how segment payloads must look like

Reply:
now, we could argue wether the yaml.ls coupling to segment format is a good or a bad thing